onert-micro/onert-micro/include/onert-micro-train.h

```diff
@@ -0,0 +1,235 @@
+
+float training_input[BATCH_SIZE*INPUT_SIZE];
+float training_label[BATCH_SIZE*OUTPUT_SIZE];
+
+// main training loop
+for(int epoch=0; epoch < NUM_EPOCHS; epoch++) {
+  for(int step=0; step < NUM_BATCHES ; step++) {
+    // prepare this steps's intput/label
+    memcpy(training_input, train_input_data + THIS_BATCH_OFFSET, BATCH_SIZE*INPUT_SIZE);
+    memcpy(training_output, train_output_data + THIS_BATCH_OFFSET, BATCH_SIZE*OUTPUT_SIZE);
+
+    // 2. feed training input / output
+    om_train_set_input(ctx, 0 , training_input, BATCH_SIZE*INPUT_SIZE);
+    om_train_set_expected(ctx, 0 , training_input,BATCH_SIZE*INPUT_SIZE);
+    // 3. train a step
+    om_train_single_step(ctx);
+  }
+  // 4. check loss
+  float loss;
+  om_train_get_loss(ctx, 0, &loss);
+
+  if(loss > TARGET_LOSS) {
+    om_train_save_as_checkpoint(ctx, PATH);
+  }
+  else {
+    om_train_save_as_inferencemodel(ctx, PATH);
+  }
+}
+*/
+
+
+
+
+/**
+ * @brief Result values returned from a call to an API function
+ */
+typedef enum
+{
+  /** Successful */
+  OM_STATUS_NO_ERROR = 0,
+  /**
+   * An error code for general use.
+   * Mostly used when there is no specific value for that certain situation.
+   */
+  OM_STATUS_ERROR = 1,
+} OM_STATUS;
+
+struct om_context;
+
+/**
+ * @brief Create a new context.
+ *
+ * @param[out]  cxt The context to be created
+ * @return      OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_create_context(om_context **cxt);
+
+/**
+ * @brief Load model from model dir
+ *
+ * @param[in] ctx  context loading the given model dir
+ * @param[in] path Path to the model or to the dir which contains training artifacts
+ *
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_load_model_from_file(om_context *ctx, const char *path);
+
+/**
+ * @brief Load checkpoint from checkpoint dir
+ *
+ * @param[in] ctx  context loading the given checkpoint
+ * @param[in] path Path to the checkpoint dir which contains all the checkpoint files
+ *
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_load_checkpoint_from_file(om_context *ctx, const char *path);
+
+
+/**
+ * @brief Prepare context to be ready for training
+ * @note  The context will be entered into training mode
+ *
+ * @param[in] ctx The context to be prepared for training
+ *
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_train_compile(om_context *ctx);
+
+
+/**
+ * @brief Set training input
+ * @note  This function should be called after {@link om_train_compile}
+ *
+ * @param[in] context     The context to be set training inputs and expected model outputs
+ * @param[in] index       The index of training input
+ * @param[in] input       The input buffers for training
+ * @param[in] size        The byte size of input buffer
+ *                        If it is nullptr, it will not change shape and batch size
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_train_set_input(om_context *ctx, uint32_t index, const void *input, int size);
```

Comment:
can be conbined with om_train_inference, I think.

Reply:
As mentioned before, I'd like to keep one api for one role. Thus, hope that this api will be as it is. Only different between you suggestion and current function is single `om_train_set_input` call